x-pack/legacy/plugins/actions/server/lib/execute.ts

```diff
@@ -25,27 +30,32 @@ export async function execute({
   params,
   encryptedSavedObjectsPlugin,
 }: ExecuteOptions): Promise<ActionTypeExecutorResult> {
-  // TODO: Ensure user can read the action before processing
-  const action = await encryptedSavedObjectsPlugin.getDecryptedAsInternalUser('action', actionId, {
+  // Ensure user can read the action before processing
+  const {
+    attributes: { actionTypeId, config, description },
+  } = await services.savedObjectsClient.get<RawAction>('action', actionId);
+  // Only get encrypted attributes here, the remaining attributes can be fetched in
+  // the savedObjectsClient call
+  const {
+    attributes: { secrets },
+  } = await encryptedSavedObjectsPlugin.getDecryptedAsInternalUser<RawAction>('action', actionId, {
     namespace,
   });
-  const actionType = actionTypeRegistry.get(action.attributes.actionTypeId);
+  const actionType = actionTypeRegistry.get(actionTypeId);
 
   let validatedParams;
   let validatedConfig;
   let validatedSecrets;
 
   try {
     validatedParams = validateParams(actionType, params);
-    validatedConfig = validateConfig(actionType, action.attributes.config);
-    validatedSecrets = validateSecrets(actionType, action.attributes.secrets);
+    validatedConfig = validateConfig(actionType, config);
+    validatedSecrets = validateSecrets(actionType, secrets);
```

Comment:
Are we at risk of leaking the secret values here if they happen to fail the validation?

Reply:
The validators use `@kbn/config-schema` and from what I understand it doesn't leak data when validating. @pmuellr can you confirm that is the case?